`Y020` false-positive for `__slots__`

This code 

```python
class CommandsParser:
    __slots__ = ("commands", "node")
```

produces

```
./stubs/redis/redis/asyncio/parser.pyi:7:18: Y020 Quoted annotations should never be used in stubs
./stubs/redis/redis/asyncio/parser.pyi:7:30: Y020 Quoted annotations should never be used in stubs
```

I don't think it is correct, because this is how `__slots__` should be defined.

Refs https://github.com/python/typeshed/pull/8830

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -5,6 +5,10 @@
 Bugfixes:
 * Do not emit Y020 for empty strings. Y020 concerns "quoted annotations",
   but an empty string can never be a quoted annotation.
+* Add special-casing so that Y020 is not emitted for `__slots__` definitions
+  inside `class` blocks.
+* Expand Y035 to cover `__slots__` definitions as well as `__match_args__` and
+  `__all__` definitions.
 
 ## 22.8.2
 

--- a/pyi.py
+++ b/pyi.py
@@ -633,7 +633,7 @@ def _non_kw_only_args_of(args: ast.arguments) -> list[ast.arg]:
 def _is_assignment_which_must_have_a_value(
     target_name: str | None, *, in_class: bool
 ) -> bool:
-    return (target_name == "__match_args__" and in_class) or (
+    return (target_name in {"__match_args__", "__slots__"} and in_class) or (
         target_name == "__all__" and not in_class
     )
 

--- a/tests/names_requiring_values.pyi
+++ b/tests/names_requiring_values.pyi
@@ -2,6 +2,7 @@ __all__: list[str]  # Y035 "__all__" in a stub file must have a value, as it has
 __all__: list[str] = ["foo", "bar", "baz"]
 __all__ = ["foo", "bar", "baz"]
 __match_args__: list[int]
+__slots__: set[int]
 
 foo: int = ...
 bar: str = ...
@@ -10,6 +11,7 @@ baz: list[set[bytes]] = ...
 class Foo:
     __all__: list[str]
     __match_args__: tuple[str, ...]  # Y035 "__match_args__" in a stub file must have a value, as it has the same semantics as "__match_args__" at runtime.
+    __slots__: tuple[str, ...]  # Y035 "__slots__" in a stub file must have a value, as it has the same semantics as "__slots__" at runtime.
 
 class Bar:
     __match_args__ = ('x', 'y')

--- a/tests/quotes.pyi
+++ b/tests/quotes.pyi
@@ -10,6 +10,7 @@ __all__.extend(["i"])
 __all__.append("j")
 __all__.remove("j")
 __match_args__ = ('foo',)  # Y020 Quoted annotations should never be used in stubs
+__slots__ = ('foo',)  # Y020 Quoted annotations should never be used in stubs
 
 def f(x: "int"): ...  # Y020 Quoted annotations should never be used in stubs
 def g(x: list["int"]): ...  # Y020 Quoted annotations should never be used in stubs
@@ -27,6 +28,7 @@ class Child(list["int"]):  # Y020 Quoted annotations should never be used in stu
 
     __all__ = ('foo',)  # Y020 Quoted annotations should never be used in stubs
     __match_args__ = ('foo', 'bar')
+    __slots__ = ('foo', 'bar')
 
 if sys.platform == "linux":
     f: "int"  # Y020 Quoted annotations should never be used in stubs

